// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package kv_server.v2;

import "google/api/annotations.proto";
import "google/api/httpbody.proto";

// Key Value Service API V2.
// Note: V2 API is still in an experimental state. Please use the V1 API.
service KeyValueService {
  // Plain text version of the V2 GetValues query API. This is intended to be
  // used for debugging and demoing purposes. The result is a JSON dictionary
  // where the key is the compression group of the partitions and the value
  // contains the output for the requested partitions.
  //
  // TODO(b/263284614): Add more comments once the explainer is published.
  rpc GetValues(GetValuesRequest) returns (google.api.HttpBody) {
    option (google.api.http) = {
      put: "/v2/getvalues"
      body: "raw_body"
    };
  }
}

message GetValuesRequest {
  google.api.HttpBody raw_body = 1;
}
